backend/src/v2/component/launcher_v2.go

```diff
@@ -148,6 +148,17 @@ func (l *LauncherV2) Execute(ctx context.Context) (err error) {
 			}
 		}
 		glog.Infof("publish success.")
+		// At the end of the current task, we check the statuses of all tasks in the current DAG and update the DAG's status accordingly.
+		// TODO: If there's a pipeline whose only components are DAGs, this launcher logic will never run and as a result the dag status will never be updated. We need to implement a mechanism to handle this edge case.
```

Comment:
> If there's a pipeline whose only components are DAGs

I can't imagine a valid case for this. and I vaguely recall the compiler check that a pipeline must have concrete tasks.

Reply:
If that's the case then I'll reword it, as the screenshots of the `oneof` scenario above showcases that there are some instances of dags that seem to not be updated with this functionality. There probably is some logic that can be used to address those edge cases.